dfastmi/batch/AreaPlotter.py

```diff
@@ -0,0 +1,388 @@
+                areai,
+                wbin,
+                wbin_labels,
+                wthresh,
+                siface,
+                afrac,
+                sbin,
+                sthresh,
+                kmid,
+            )
+
+    def _plot_figures_with_details_for_n_areas_with_largest_volumes(
+        self,
+        dzgemi: numpy.ndarray,
+        areai: numpy.ndarray,
+        wbin: numpy.ndarray,
+        wbin_labels: list[str],
+        wthresh: numpy.ndarray,
+        siface: numpy.ndarray,
+        afrac: numpy.ndarray,
+        sbin: numpy.ndarray,
+        sthresh: numpy.ndarray,
+        kmid: numpy.ndarray,
+    ):
+        volume_mean = self._area_detector.volume[1:, :].mean(axis=0)
+        sorted_list = numpy.argsort(volume_mean)[::-1]
+        if len(sorted_list) <= self._plot_n:
+            vol_thresh = 0.0
+        else:
+            vol_thresh = volume_mean[sorted_list[self._plot_n]]
+
+        self._plot_certain_areas(
+            volume_mean > vol_thresh,
+            dzgemi,
+            areai,
+            wbin,
+            wbin_labels,
+            siface,
+            afrac,
+            sbin,
+            wthresh,
+            sthresh,
+            kmid,
+        )
+
+    def _plot_certain_areas(
+        self,
+        condition: bool,
+        dzgemi: numpy.ndarray,
+        areai: numpy.ndarray,
+        wbin: numpy.ndarray,
+        wbin_labels: list[str],
+        siface: numpy.ndarray,
+        afrac: numpy.ndarray,
+        sbin: numpy.ndarray,
+        wthresh: numpy.ndarray,
+        sthresh: numpy.ndarray,
+        kmid: numpy.ndarray,
+    ):
+        indices = numpy.nonzero(condition)[0]
+        sbin_length = sthresh[1] - sthresh[0]
+        for ia in indices:
+            dzgemi_filtered = dzgemi.copy()
+            dzgemi_filtered[numpy.invert(self._area_detector.area_list[ia])] = 0.0
+
+            area_binvol = self._comp_binned_volumes(
+                dzgemi_filtered, areai, wbin, siface, afrac, sbin, wthresh, sthresh
+            )
+
+            fig, ax = dfastmi.plotting.plot_sedimentation(
+                kmid,
+                "chainage [km]",
+                area_binvol,
+                "volume [m3] accumulated per {} m bin alongstream".format(sbin_length),
+                self._area_str.format(ia + 1),
+                wbin_labels,
+                positive_up=self._positive_up,
+            )
+
+            figure_base_name = (
+                self._area_str.replace(" ", "_").format(ia + 1) + "_volumes"
+            )
+            self._save_figure(fig, ax, figure_base_name)
+
+    def _save_figure(self, fig: Figure, ax: Axes, figure_base_name: str):
+        if not self._plotting_options.saveplot:
+            return
+
+        figbase = self._plotting_options.figure_save_directory / figure_base_name
+        if self._plotting_options.saveplot_zoomed:
+            dfastmi.plotting.zoom_x_and_save(
+                fig,
+                ax,
+                figbase,
```

Comment:
You are passing a path, while a string is expected.

Reply:
![1csU](https://github.com/Deltares/D-FAST_Morphological_Impact/assets/102953403/6ff8040c-fa25-4936-8848-6bb4ac09ff0f)

Reply:
![gif](https://github.com/Deltares/D-FAST_Morphological_Impact/assets/102953403/bb8713c2-60a1-46e0-89a1-49e515ba1cb3)

Reply:
Do we want to apply path on plotting or str on the call of the path methods for now?

Reply:
My preference is to have the Path object as long as possible. However, that may be out of scope for this issue.